internal/engine/container_build_and_deployer.go

```diff
@@ -65,29 +66,31 @@ func (cbd *ContainerBuildAndDeployer) BuildAndDeploy(ctx context.Context, servic
 	if state.LastResult.Container.String() == "" {
 		pID, err := cbd.k8sClient.PodWithImage(ctx, state.LastResult.Image)
 		if err != nil {
-			logger.Get(ctx).Infof("Unable to find pod, falling back to image deploy: %s", err.Error())
+			logger.Get(ctx).Infof("Unable to find pod, falling back to image deploy: %s", err)
 			return cbd.ibd.BuildAndDeploy(ctx, service, state)
 		}
 		logger.Get(ctx).Infof("Deploying to pod: %s", pID)
 		// get containerID from pID (see container_updater.go --> containerIdForPod)
 		cID, err = cbd.cu.ContainerIDForPod(ctx, pID)
 		if err != nil {
-			logger.Get(ctx).Infof("Unable to find container, falling back to image deploy: %s", err.Error())
+			logger.Get(ctx).Infof("Unable to find container, falling back to image deploy: %s", err)
 			return cbd.ibd.BuildAndDeploy(ctx, service, state)
 		}
-		logger.Get(ctx).Infof("Got container ID for pod: %s", cID)
+		logger.Get(ctx).Infof("Got container ID for pod: %s", cID.ShortStr())
 	} else {
 		cID = state.LastResult.Container
 	}
 	// once have cID -- can call cbd.cu.UpdateContainer(...)
 	cf, err := build.FilesToPathMappings(state.FilesChanged(), service.Mounts)
 	if err != nil {
-		return BuildResult{}, err
+		logger.Get(ctx).Infof("Unable to translate changed files to path mappings, falling back to image deploy: %s", err)
```

Comment:
don't we need to do this same step in an image deploy? i don't think this is a good case to fall back in

Reply:
yes good call